### PR TITLE
EZQMS-602: Moved Rank to its own package

### PR DESCRIPTION
**Pull request by anotherpit**
# Contribution checklist

## Brief description

Moved Rank to core

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [ ] - Does a local formatting is applied (rush format)
* [ ] - Does a local svelte-check is applied (rush svelte-check)
* [ ] - Does a local UI tests are executed [UI Testing](../tests/readme.md)
* [ ] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

## Contributor requirements

* [x] - Sign-off is provided. [DCO](https://github.com/apps/dco)
* [x] - GPG Signature is provided. [GPG](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)

<sub>View in Huly <a href="https://front.hc.engineering/guest/platform?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NWUxMDIyZGU5ZDA1MzJmMzJjMmRhZDIiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InBsYXRmb3JtIiwicHJvZHVjdElkIjoiIn0.DmpO4fgSFvTrHNl8hOP7ouqhhz6fcJFrSlUuDt9lNhk">UBERF-5836</a></sub>

### CODE REVIEW

**Comment by haiodo** [comment]
@anotherpit Why we need it in core?

**Review by aonnikov — DISMISSED** [review]

**Comment by anotherpit** [comment]
@haiodo Because, as stated in the issue, the `Rank` type and related utilities are already being used in 3+ modules (and currently they are re-implemented in each of them)

**Review by haiodo — CHANGES_REQUESTED** [review]
discussed to have separate package

File: plugins/task/src/utils.ts [hunk 1]
```diff
@@ -29,37 +29,18 @@ import core, {
...
-    res.push(new LexoRank(LexoRankBucket.BUCKET_0, cur).toString())
-  }
-  return res
-}
+export { genRanks, makeRank } from '@hcengineering/rank'
 
 /**
  * @public
+ *
+ * TODO: Drop after everything migrates to {@link makeRank}
```
Comment (anotherpit) [inline plugins/task/src/utils.ts:40]: Need to migrate `uberflow` first, and then drop it here

**Review by aonnikov — APPROVED** [review]

File: plugins/task/src/utils.ts (hunk 1 above)
Comment (haiodo) [reply plugins/task/src/utils.ts:40]: Please do not drop until all 3 our deployments will be migrated.

**Review by haiodo — APPROVED** [review]

**Pull request merged**